pcsx2/Input/DualShock3InputSource.h

```diff
@@ -0,0 +1,168 @@
+	void UpdateSettings(SettingsInterface& si, std::unique_lock<std::mutex>& settings_lock) override;
+	bool ReloadDevices() override;
+	void Shutdown() override;
+
+	void PollEvents() override;
+	std::vector<std::pair<std::string, std::string>> EnumerateDevices() override;
+	std::vector<InputBindingKey> EnumerateMotors() override;
+	bool GetGenericBindingMapping(const std::string_view& device, InputManager::GenericInputBindingMapping* mapping) override;
+	void UpdateMotorState(InputBindingKey key, float intensity) override;
+	void UpdateMotorState(InputBindingKey large_key, InputBindingKey small_key, float large_intensity, float small_intensity) override;
+
+	std::optional<InputBindingKey> ParseKeyString(const std::string_view& device, const std::string_view& binding) override;
+	std::string ConvertKeyToString(InputBindingKey key) override;
+
+private:
+
+	//Sixaxis driver commands.
+	//All commands must be sent via WriteFile with 49-byte buffer containing output report.
+	//Byte 0 indicates reportId and must always be 0.
+	//Byte 1 indicates some command, supported values are specified below.
+	enum SIXASIS_DRIVER_COMMANDS : u8
+	{
+		//This command allows to set user LEDs.
+		//Bytes 5,6.7.8 contain mode for corresponding LED: 0 value means LED is OFF, 1 means LEDs in ON and 2 means LEDs is flashing.
+		//Bytes 9-16 specify 64-bit LED flash period in 100 ns units if some LED is flashing, otherwise not used.
+		SIXAXIS_COMMAND_SET_LEDS = 1,
+
+		//This command allows to set left and right motors.
+		//Byte 5 is right motor duration (0-255) and byte 6, if not zero, activates right motor. Zero value disables right motor.
+		//Byte 7 is left motor duration (0-255) and byte 8 is left motor amplitude (0-255).
+		SIXAXIS_COMMAND_SET_MOTORS = 2, 
+
+		//This command allows to block/unblock setting device LEDs by applications.
+		//Byte 5 is used as parameter - any non-zero value blocks LEDs, zero value will unblock LEDs.
+		SIXAXIS_COMMAND_BLOCK_LEDS = 3,
+
+		//This command refreshes driver settings. No parameters used.
+		//When sixaxis driver loads it reads 'CurrentDriverSetting' binary value from 'HKLM\System\CurrentControlSet\Services\sixaxis\Parameters' registry key.
+		//If the key is not present then default values are used. Sending this command forces sixaxis driver to re-read the registry and update driver settings.
+		SIXAXIS_COMMAND_REFRESH_DRIVER_SETTING = 9,
+
+		//This command clears current bluetooth pairing. No parameters used.
+		SIXAXIS_COMMAND_CLEAR_PAIRING = 10
+	};
+	
+	struct DS3ControllerData
+	{
+		int player_id;
+		std::wstring device_path; 
+
+		HANDLE hFile = INVALID_HANDLE_VALUE;		
+
+		bool active = false;
+
+		u8 SmallMotorOn = 0; // 0 or 1 (off/on)
+		u8 LargeMotorForce = 0; // range [0, 255]
+		
+		std::array<bool, 17> physicalButtonState = {}; 
+		std::array<bool, 17> lastPhysicalButtonState = {};
+
+		//If we wever want to support pressure sensitive buttons
```

Comment:
Seems a bit silly to go to all the effort of adding a DS3-specific backend but not support pressure-sensitive buttons? Since that would be the main reason for using a DS3.

May as well just use DInput in that case.